Notebook.ipynb

```diff
@@ -334,22 +364,88 @@
   },
   {
    "cell_type": "markdown",
-   "id": "7b95e398",
+   "id": "ff5e667b",
+   "metadata": {},
+   "source": [
+    "To face the variability induced by the fact that every element in the map (carrots, saddle, horse, monsters and sometimes even the agent) spawns randomly, we iterated many tests and made our assessment over their results. \\\n",
+    "The general behaviour of the agent has been tested over the levels 0, 1, 2, 3 and -1. The latter is a level where the agent is randomly put in a randomly generated maze, level 0 represents a small, rectangular room, level 1 is composed by two different rooms, connected by a narrow corridor, level 2 is a level that more closely resembles a NetHack level, and finally level 3 which is a fairly complex maze with narrow corridors.\\\n",
+    "Furthermore, we evaluated the behavior of our agent under varying degrees of problem relaxations by testing it both with and without the contingencies presented by hunger and monsters.\n",
+    "\n",
+    "By running the cell below with the appropriate parameters it is possible to see two histograms, one indicating the medium steps on a success and failure scenario, and the other indicating the success rate of our agent in a certain level with certain conditions.\n",
+    "The naming convention we devised for the different evaluation environments is given by the concatenation of these elements:\n",
+    "- L\n",
+    "- a number between -1 and 3 to indicate the level\n",
+    "- Y or N to denote either the presence or the absence of a certain contingency \n",
+    "- A, which is the initial of \"Apples\" (eaten by our agent): denotes the contingency of hunger \n",
+    "- Y or N to indicate either the presence or the absence of a certain contingency\n",
+    "- E, which is the initial of \"Enemies\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1df0a316-4a1b-4ef9-bc29-a0b4ecdedd17",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from stats.show_stats import show_stats\n",
+    "\n",
+    "# -1 for random maze\n",
+    "# 0 for square\n",
+    "# 1 for impossible one\n",
+    "# 2 for nethack-like level\n",
+    "# 3 for complex maze\n",
+    "\n",
+    "# success_rate_flag must be set to True to see the rate of success of the agent\n",
```

Comment:
i would talk somewhere around here that these numbers come from a 200-2000 times executions